integration_test/tests/common/mod.rs

```diff
@@ -126,30 +121,50 @@ pub fn sql_return_to_string(return_code: SqlReturn) -> String {
 }
 
 #[allow(dead_code)]
-// Allocates new environment variable
-pub fn allocate_env() -> Result<HEnv> {
+/// Setup flow that connects and allocates a statement. This allocates a new
+/// environment handle, sets the ODBC_VERSION environment attribute, connects
+/// using the provided (or default) URI, and allocates a statement. The flow
+/// is:
+///     - SQLAllocHandle(SQL_HANDLE_ENV)
+///     - SQLSetEnvAttr(SQL_ATTR_ODBC_VERSION, SQL_OV_ODBC3)
+///     - SQLAllocHandle(SQL_HANDLE_DBC)
+///     - SQLDriverConnectW
+///     - SQLAllocHandle(SQL_HANDLE_STMT)
+pub fn default_setup_connect_and_alloc_stmt(
+    odbc_version_value: AttrOdbcVersion,
+) -> (HEnv, HDbc, HStmt) {
+    let env_handle = allocate_env(odbc_version_value);
+    let (conn_handle, stmt_handle) = connect_and_allocate_statement(env_handle, None);
+
+    (env_handle, conn_handle, stmt_handle)
+}
+
+#[allow(dead_code)]
+/// Allocates new environment variable and sets the ODBC_VERSION environment
+/// attribute to the provided odbc_version.
+pub fn allocate_env(odbc_version: AttrOdbcVersion) -> HEnv {
```

Comment:
[nit] We only to set the odbc version to ODBC 2.0 in one or 2 places.
I would have left the existing as-is instead of forcing us to pass in SQL_OV_ODBC3 in every call now.
What was bothering you with this function?

Reply:
ah, I see now, yes makes sense! 👍